bool control
int32 shape_id
float64 center_x
float64 center_y
float64 area
float64 direction